stan/math/rev/functor/algebra_solver_powell.hpp

```diff
@@ -15,168 +18,117 @@
- *
- * Throw an exception if the norm of f(x), where f is the
- * output of the algebraic system and x the proposed solution,
- * is greater than the function tolerance. We here use the
- * norm as a metric to measure how far we are from the origin (0).
- *
- * @tparam F type of equation system function.
- * @tparam T type of initial guess vector.
- *
- * @param[in] f Functor that evaluates the system of equations.
- * @param[in] x Vector of starting values.
- * @param[in] y parameter vector for the equation system. The function
- *            is overloaded to treat y as a vector of doubles or of a
- *            a template type T.
- * @param[in] dat continuous data vector for the equation system.
- * @param[in] dat_int integer data vector for the equation system.
- * @param[in, out] msgs the print stream for warning messages.
- * @param[in] relative_tolerance determines the convergence criteria
- *            for the solution.
- * @param[in] function_tolerance determines whether roots are acceptable.
- * @param[in] max_num_steps  maximum number of function evaluations.
- * @return theta Vector of solutions to the system of equations.
- * @throw <code>std::invalid_argument</code> if x has size zero.
- * @throw <code>std::invalid_argument</code> if x has non-finite elements.
- * @throw <code>std::invalid_argument</code> if y has non-finite elements.
- * @throw <code>std::invalid_argument</code> if dat has non-finite elements.
- * @throw <code>std::invalid_argument</code> if dat_int has non-finite elements.
- * @throw <code>std::invalid_argument</code> if relative_tolerance is strictly
- * negative.
- * @throw <code>std::invalid_argument</code> if function_tolerance is strictly
- * negative.
- * @throw <code>std::invalid_argument</code> if max_num_steps is not positive.
- * @throw <code>std::domain_error</code> solver exceeds max_num_steps.
- * @throw <code>std::domain_error</code> if the norm of the solution exceeds
- * the function tolerance.
- */
-template <typename F, typename T, require_eigen_vector_t<T>* = nullptr>
-Eigen::VectorXd algebra_solver_powell(
-    const F& f, const T& x, const Eigen::VectorXd& y,
-    const std::vector<double>& dat, const std::vector<int>& dat_int,
-    std::ostream* msgs = nullptr, double relative_tolerance = 1e-10,
-    double function_tolerance = 1e-6,
-    long int max_num_steps = 1e+3) {  // NOLINT(runtime/int)
-  const auto& x_eval = x.eval();
-  const auto& x_val = (value_of(x_eval)).eval();
-  algebra_solver_check(x_val, y, dat, dat_int, function_tolerance,
-                       max_num_steps);
-  check_nonnegative("alegbra_solver", "relative_tolerance", relative_tolerance);
-  // if (relative_tolerance < 0)
-  //   invalid_argument("algebra_solver", "relative_tolerance,",
-  //                    function_tolerance, "",
-  //                    ", must be greater than or equal to 0");
+  // Solve the system
+  return algebra_solver_powell_(solver, fx, x_val, 0, relative_tolerance,
```

Comment:
Give this a more meaningful name. We already used `algebra_solver_powell_impl` too lol. Something very verbose is fine if it's easier to come up with something.